packages/xml-views-validation/test/validators/document/non-unique-id-spec.ts

```diff
@@ -0,0 +1,170 @@
+import { expect } from "chai";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { generateModel } from "@ui5-language-assistant/test-utils";
+import { validateNonUniqueID } from "../../../src/validators/document/non-unique-id";
+import { NonUniqueIDIssue } from "../../../api";
+import {
+  computeExpectedRanges,
+  testValidationsScenario,
+} from "../../test-utils";
+
+describe("the use of non unique id validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  let testNonUniqueIDScenario: (opts: {
+    xmlText: string;
+    assertion: (issues: NonUniqueIDIssue[]) => void;
+  }) => void;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({ version: "1.74.0" });
+
+    testNonUniqueIDScenario = (opts): void =>
+      testValidationsScenario({
+        model: ui5SemanticModel,
+        validators: {
+          document: [validateNonUniqueID],
+        },
+        xmlText: opts.xmlText,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        assertion: opts.assertion as any,
+      });
+  });
+
+  context("true positive scenarios", () => {
```

Comment:
Add a test where there are 3 of the same ID, and a test where there are 2 non-unique IDs in the document

Reply:
3 of the same ID is a good use case to test.

2 separate duplicate IDs less so, because then I would be testing the validation framework or even more precisely the ASTVisitor. so I don't think its the responsibility of a specific validation tests to prove they run on all relevant AST Nodes.

Maybe we could expend a smoke test here:
- https://github.com/SAP/ui5-language-assistant/blob/master/packages/xml-views-validation/test/api-spec.ts

- But it already shows 2 separate validations executed on two separate nodes.

Reply:
Hmm maybe it would indeed be useful to have multiple IDs test because there is a custom logic here to collect the IDs

Reply:
Also I think there should be a test for duplicate ID in custom control (unknown namespace with tag starting with uppercase)

Reply:
> Also I think there should be a test for duplicate ID in custom control (unknown namespace with tag starting with uppercase)

Yes, I'll implement this